consensus/enclave/impl/src/lib.rs

```diff
@@ -422,6 +426,20 @@ impl ConsensusEnclave for SgxConsensusEnclave {
             }
         }
 
+        // Duplicate output public keys are not allowed.
+        let mut seen_output_public_keys = BTreeSet::default();
```

Comment:
Why is this a BTreeSet (instead of a HashSet)?

Reply:
I'm wondering if BTreeSet would be significantly slower than HashSet. Hard to tell:

https://doc.rust-lang.org/std/collections/struct.BTreeMap.html

"Currently, our implementation simply performs naive linear search. This provides excellent performance on small nodes of elements which are cheap to compare. However in the future we would like to further explore choosing the optimal search strategy based on the choice of B, and possibly other factors. Using linear search, searching for a random element is expected to take O(B * log(n)) comparisons, which is generally worse than a BST. In practice, however, performance is excellent."

Reply:
Quicker lookups, similar to why `used_key_images` is also a BTreeSet. Insertion is slower, but you are doing N insertions and N^2 lookups (I think).

Reply:
Work that was done a long time ago on optimizing performance yielded better results when using BTreeSets.

Reply:
Ok, that makes sense.